src/store/aggregator/getIncentive.ts

```diff
@@ -0,0 +1,76 @@
+import { Request, newHttpClient } from 'typescript-http-client';
+import Web3 from 'web3';
+import { AbiItem } from 'web3-utils';
+import { BigNumber } from 'ethers';
+import {
+  getPenaltyForWithdrawalAbi,
+  getRewardForDepositAbi,
+} from './sagas/rewardManagerAbis';
+import { ChainEnum } from '../../config/chains';
+import { contractsAddresses } from '../../config/contracts';
+
+const publicTestnetNodeUrl = 'https://public-node.testnet.rsk.co';
+const publicMainnetBodeUrl = 'https://public-node.rsk.co';
+
+const web3 = new Web3();
+const client = newHttpClient();
+
+interface JsonRpcResponse {
+  result: string;
+}
+
+interface JsonRpcRequest {}
+
+async function jsonRpcPost(
+  mainnetFlag: boolean,
+  data: JsonRpcRequest
+): Promise<JsonRpcResponse> {
+  const url = mainnetFlag ? publicMainnetBodeUrl : publicTestnetNodeUrl;
+  const rewardManagerAddress =
+    contractsAddresses[mainnetFlag ? ChainEnum.RSK : ChainEnum.RSK_TESTNET]
+      .rewardManager;
+  const request = new Request(url, {
+    method: 'POST',
+    responseType: 'json',
+    body: {
+      jsonrpc: '2.0',
+      id: `${new Date()}`,
+      method: 'eth_call',
+      params: [
+        {
+          to: rewardManagerAddress,
+          data,
+        },
+        'latest',
+      ],
+    },
+  });
+  const responseBody = await client.execute<JsonRpcResponse>(request);
+  return responseBody;
+}
+
+export async function getReward(
```

Comment:
This hack is probably not needed anymore, fix all my other comments first and then try to make it working as I described in this comment: https://sovryn.atlassian.net/browse/SOV-1309?focusedCommentId=17176.

Reply:
We are no longer in a position to spend time changing things to no functional advantage. This works. If you think that the other way is "less hacky", we should do that as a separate ticket.

Reply:
Why do you say we're in a rush now? @john-light can you confirm we should go with the current approach even though it means we'll introduce some technical debt? I think we should do our best and not make the codebase worse than already is.

Reply:
@john-light is our product manager, not our technical decision maker. As this is a technical decision, I think I'll make the call. I don't disagree with you, I just think that this is an improvement that can and should be put in a ticket and done as a separate PR. The reasons are: 1. time is short, 2. the current solution is localized, clear, and tested, and 3. there is technical debt in this project that should be fixed and is much higher priority - specifically the failing tests, low test coverage, and convoluted logic.

Reply:
> @john-light is our product manager, not our technical decision maker. As this is a technical decision, I think I'll make the call.

While I appreciate the consideration to ask for my opinion on how this should be implemented, as long as the requirements in the PRD/product spec are satisfied it is @dharkmattr's role to make the final call on _how_ to technically satisfy the requirements. So on this I will defer to his decision.